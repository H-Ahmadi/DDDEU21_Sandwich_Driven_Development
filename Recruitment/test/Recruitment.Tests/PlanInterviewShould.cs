using System;
using System.Collections.Generic;
using ApprovalTests.Reporters;
using ApprovalTests.Reporters.TestFrameworks;
using Recruitment.Application;
using Recruitment.Domain.Candidates;
using Recruitment.Domain.Interviews;
using Recruitment.Domain.Recruiters;
using Recruitment.Tests.TestDoubles;
using Xunit;

namespace Recruitment.Tests
{
    [UseReporter(typeof(XUnit2Reporter))]
    public class PlanInterviewShould
    {
        private readonly ICandidateRepository _candidates;
        private readonly IInterviewRepository _interviews;
        private readonly IRecruiterRepository _recruiters;
        public PlanInterviewShould()
        {
            _candidates = new FakeCandidates();
            _interviews = new FakeInterviews();
            _recruiters = new FakeRecruiters();
        }

        [Fact]
        public void find_first_available_recruiter_who_can_test_candidate()
        {
            ApprovalTests.Combinations.CombinationApprovals.VerifyAllCombinations(
                PlanInterview,
                new List<long>() { 123, 456, 789, 10000, 20000, 30000 },
                new List<DateTime>()
                {
                    new DateTime(2021, 02, 20),
                    new DateTime(2021, 02, 21),
                    new DateTime(2021, 02, 22),
                }
            );
        }

        private Interview PlanInterview(long candidateId, DateTime availability)
        {
            var planner = new PlanInterview(_candidates, _recruiters, _interviews);
            return planner.Plan(candidateId, availability);
        }
    }
}
