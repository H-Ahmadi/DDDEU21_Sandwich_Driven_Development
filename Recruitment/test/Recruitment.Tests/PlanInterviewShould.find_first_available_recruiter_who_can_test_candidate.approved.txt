[123,02/20/2021 00:00:00] => Interview{Candidate=Candidate{Skills=.NET,Java,PHP,JS, Name='Alex'} ,Recruiter=Recruiter{Id=2, Skills=Java,.NET,PHP,JS, Availabilities=2/22/2021 12:00:00 AM, Name='Mary'} ,InterviewDate=2/20/2021 12:00:00 AM ,}
[123,02/21/2021 00:00:00] => Exception of type 'Recruitment.Domain.Recruiters.RecruiterNotFoundException' was thrown.
[123,02/22/2021 00:00:00] => Interview{Candidate=Candidate{Skills=.NET,Java,PHP,JS, Name='Alex'} ,Recruiter=Recruiter{Id=2, Skills=Java,.NET,PHP,JS, Availabilities=, Name='Mary'} ,InterviewDate=2/22/2021 12:00:00 AM ,}
[456,02/20/2021 00:00:00] => Interview{Candidate=Candidate{Skills=JS, Name='Bob'} ,Recruiter=Recruiter{Id=1, Skills=PHP,JS, Availabilities=, Name='Emma'} ,InterviewDate=2/20/2021 12:00:00 AM ,}
[456,02/21/2021 00:00:00] => Exception of type 'Recruitment.Domain.Recruiters.RecruiterNotFoundException' was thrown.
[456,02/22/2021 00:00:00] => Exception of type 'Recruitment.Domain.Recruiters.RecruiterNotFoundException' was thrown.
[789,02/20/2021 00:00:00] => Exception of type 'Recruitment.Domain.Recruiters.RecruiterNotFoundException' was thrown.
[789,02/21/2021 00:00:00] => Exception of type 'Recruitment.Domain.Recruiters.RecruiterNotFoundException' was thrown.
[789,02/22/2021 00:00:00] => Exception of type 'Recruitment.Domain.Recruiters.RecruiterNotFoundException' was thrown.
[10000,02/20/2021 00:00:00] => Object reference not set to an instance of an object.
[10000,02/21/2021 00:00:00] => Exception of type 'Recruitment.Domain.Recruiters.RecruiterNotFoundException' was thrown.
[10000,02/22/2021 00:00:00] => Exception of type 'Recruitment.Domain.Recruiters.RecruiterNotFoundException' was thrown.
[20000,02/20/2021 00:00:00] => Object reference not set to an instance of an object.
[20000,02/21/2021 00:00:00] => Exception of type 'Recruitment.Domain.Recruiters.RecruiterNotFoundException' was thrown.
[20000,02/22/2021 00:00:00] => Exception of type 'Recruitment.Domain.Recruiters.RecruiterNotFoundException' was thrown.
[30000,02/20/2021 00:00:00] => Object reference not set to an instance of an object.
[30000,02/21/2021 00:00:00] => Exception of type 'Recruitment.Domain.Recruiters.RecruiterNotFoundException' was thrown.
[30000,02/22/2021 00:00:00] => Exception of type 'Recruitment.Domain.Recruiters.RecruiterNotFoundException' was thrown.
